metric_providers/powermetrics/provider.py

```diff
@@ -35,43 +41,44 @@ def __init__(self, resolution, skip_check=False):
             '-o',
             self._filename]
 
+
     def check_system(self):
-            if self.is_powermetrics_running():
-                raise MetricProviderConfigurationError('Another instance of powermetrics is already running on the system!\nPlease close it before running the Green Metrics Tool.')
+        if self._pm_process_count > 0:
+            raise MetricProviderConfigurationError('Another instance of powermetrics is already running on the system!\n'
+                                                   'Please close it before running the Green Metrics Tool.\n'
+                                                   'You can also override this with --skip-system-checks\n')
 
-    def is_powermetrics_running(self):
-        ps = subprocess.run(['pgrep', '-qx', 'powermetrics'], check=False)
-        if ps.returncode == 1:
-            return False
-        return True
+    def is_our_powermetrics_running(self):
+        pw_processes = subprocess.check_output(['pgrep', '-ix', 'powermetrics'], text=True)
```

Comment:
Why using text=True here instead of encoding="UTF-8" ?

I think both do the same but I would opt for being consistent throughout the codebase.

Reply:
As far as I can see "text=True" uses the systems encoding and is preferred to forcing the encoding. In most cases UTF is the system encoding so it doesn't really matter. Maybe we should think about changing the UTF to text at some stage.

Reply:
I did a similar research when waging if going for one or the other switch. Actually what I found is that users generally prefer a UTF-8 encoding, even if the have non-classic charsets like GB* (chinese).

I would strongly prefer being forceful and explicit, rather than using system encoding ... at least until we have a breaking case.